src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -271,33 +271,6 @@ private bool CanValueBindAnyModelProperties(ModelBindingContext bindingContext)
             return false;
         }
 
-        private bool CanBindValue(ModelBindingContext bindingContext)
-        {
-            var valueProvider = bindingContext.ValueProvider;
-
-            var bindingSource = bindingContext.BindingSource;
-            if (bindingSource != null && !bindingSource.IsGreedy)
-            {
-                var rootValueProvider = bindingContext.OperationBindingContext.ValueProvider as IBindingSourceValueProvider;
-                if (rootValueProvider != null)
-                {
-                    valueProvider = rootValueProvider.Filter(bindingSource);
-                    if (valueProvider == null)
-                    {
-                        // Unable to find a value provider for this binding source. Binding will fail.
-                        return false;
-                    }
-                }
-            }
```

Comment:
Where did this block move to? Or, are you just no longer worried about the eventual binding failure?